cmd/sidecar/main.go

```
@@ -76,28 +98,29 @@ func main() {
 				return err
 			}
 
-			// ------------ START LOGIC ------------
-
-			k8sSecretsMap, err := secrets.RetrieveK8sSecrets()
-			if err != nil {
-				errLogger.Printf("Failed to retrieve k8s secrets: %s", err.Error())
-				return err
-			}
-
-			k8sSecretsMap, err = secrets.UpdateK8sSecretsMapWithConjurSecrets(k8sSecretsMap)
-			if err != nil {
-				errLogger.Printf("Failed to update K8s Secrets map: %s", err.Error())
-				return err
+			if conjurSecretsDest == 1 {
+
+				// ------------ START LOGIC ------------
+				k8sSecretsMap, err := secretsHandler.RetrieveK8sSecrets()
+				if err != nil {
+					errLogger.Printf("Failed to retrieve k8s secrets: %s", err.Error())
+					return err
+				}
+
+				k8sSecretsMap, err = secretsHandler.UpdateK8sSecretsMapWithConjurSecrets(k8sSecretsMap)
+				if err != nil {
+					errLogger.Printf("Failed to update K8s Secrets map: %s", err.Error())
+					return err
+				}
+
+				err = secretsHandler.PatchK8sSecrets(k8sSecretsMap)
+				if err != nil {
+					errLogger.Printf("Failed to patch K8s Secrets: %s", err.Error())
+					return err
+				}
+				// ------------ END LOGIC ------------
 			}
 
-			err = secrets.PatchK8sSecrets(k8sSecretsMap)
-			if err != nil {
-				errLogger.Printf("Failed to patch K8s Secrets: %s", err.Error())
-				return err
-			}
-
-			// ------------ END LOGIC ------------
-
 			if authn.Config.ContainerMode == "init" {
```

i think this should move upwards so we fail early on this

---

also it seems that we can check this on `configAuthn` instead of `authn.config`